File: docs/vsto/codesnippet/VisualBasic/trin_ribbon_choose_ribbon_4/ThisWorkbook.vb

Public Class ThisWorkbook

    Dim myCondition As Boolean
    Private Sub ThisWorkbook_Startup() Handles Me.Startup

    End Sub

    Private Sub ThisWorkbook_Shutdown() Handles Me.Shutdown

    End Sub
    '<Snippet1>
    Protected Overrides Function CreateRibbonExtensibilityObject() As  _
        Microsoft.Office.Core.IRibbonExtensibility
        If myCondition = True Then
            Return Globals.Factory.GetRibbonFactory().CreateRibbonManager _
                (New Microsoft.Office.Tools.Ribbon.IRibbonExtension() _
                     {New Ribbon1()})
        Else
            Return Globals.Factory.GetRibbonFactory().CreateRibbonManager _
                (New Microsoft.Office.Tools.Ribbon.IRibbonExtension() _
                     {New Ribbon2()})
        End If
    End Function
    '</Snippet1>
End Class

File: docs/vsto/codesnippet/CSharp/trin_ribbon_choose_ribbon_4/ThisWorkbook.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Xml.Linq;
using Microsoft.Office.Tools.Excel;
using Microsoft.VisualStudio.Tools.Applications.Runtime;
using Excel = Microsoft.Office.Interop.Excel;
using Office = Microsoft.Office.Core;

namespace Trin_Ribbon_Choose_Ribbon_4
{
    public partial class ThisWorkbook
    {
        bool myCondition = true;
        private void ThisWorkbook_Startup(object sender, System.EventArgs e)
        {
        }

        private void ThisWorkbook_Shutdown(object sender, System.EventArgs e)
        {
        }
        //<Snippet1>
        protected override Microsoft.Office.Core.IRibbonExtensibility
    CreateRibbonExtensibilityObject()
        {
            if (myCondition == true)
            {
                return Globals.Factory.GetRibbonFactory().CreateRibbonManager(
                    new Microsoft.Office.Tools.Ribbon.IRibbonExtension[] { new Ribbon1() });
            }
            else
            {
                return Globals.Factory.GetRibbonFactory().CreateRibbonManager(
                    new Microsoft.Office.Tools.Ribbon.IRibbonExtension[] { new Ribbon2() });
            }
        }
        //</Snippet1>

        #region VSTO Designer generated code

        /// <summary>
        /// Required method for Designer support - do not modify
        /// the contents of this method with the code editor.
        /// </summary>
        private void InternalStartup()
        {
            this.Startup += new System.EventHandler(ThisWorkbook_Startup);
            this.Shutdown += new System.EventHandler(ThisWorkbook_Shutdown);
        }

        #endregion

    }
}
